### PR TITLE
Moved layout code to separate classes

### PR DESCRIPTION
**Pull request by igorkorsukov**
Just moved code, no logic changes

Step by step

### CODE REVIEW

File: src/engraving/libmscore/score.h [hunk 1]
```diff
@@ -642,10 +645,6 @@ class Score : public QObject, public ScoreElement
     Excerpt* excerpt() { return _excerpt; }
     void setExcerpt(Excerpt* e) { _excerpt = e; }
 
-    System* collectSystem(LayoutContext&);
-    void layoutSystemElements(System* system, LayoutContext& lc);
-    void getNextMeasure(LayoutContext&);        // get next measure for layout
-
```
Comment (wizofaus) [inline src/engraving/libmscore/score.h:648]: Good to see score.h being cut down!

**Review by Eism — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
Causes this in MSVC 

Warning	MSB8027	Two or more files with the name of layout.cpp will produce outputs to the same location. This can lead to an incorrect build result.  The files involved are ...\src\engraving\libmscore\layout.cpp, ...\src\engraving\libmscore\layout\layout.cpp.	engraving

**Comment by Jojo-Schmitz** [comment]
and

Warning	C4458	declaration of 'firstSystem' hides class member (compiling source file ...\src\engraving\libmscore\layout\layoutcontext.cpp 1741

see declaration of 'mu::engraving::LayoutContext::firstSystem' ...\src\engraving\libmscore\layout\layoutcontext.h	58

**Comment by Jojo-Schmitz** [comment]
Excluding those 2 layout.cpp files from Unity build doesn't seem to help though.

**Pull request merged**